DO NOT MERGE [Durable Objects] Make Durable Objects a tile (#9477)
cloudflare/cloudflare-docs: content/durable-objects/platform/limits.md

---
pcx_content_type: concept
title: Limits
weight: 9
---

# Limits

Durable Objects are only available on the Workers Paid plan.

| Feature                                    | Limit                                             |
| ------------------------------------------ | --------------------------------------------------|
|  Number of objects                         | unlimited (within an account or of a given class) |

> |  Storage per account                       | 50 GB (can be raised by contacting Cloudflare)    |

Reviewer: ```suggestion
|  Storage per account                       | 50 GB (can be raised by contacting Cloudflare) <sup>1</sup>   |
```

|  Storage per class                         | unlimited                                         |
|  Storage per object                        | unlimited                                         |
|  Key size                                  | 2 KiB(2048 bytes)                                 |
|  Value size                                | 128 KiB (131072 bytes)                            |
|  Websocket message size                    | 1 MiB (only for received messages)                |
|  CPU per request                           | 30s (including websocket messages)                |

> Durable Objects scale well across Objects, but each object is inherently single-threaded. A baseline of 100 req/sec is a good floor estimate of the request rate an individual Object can handle, though this will vary with workload.

Reviewer: ```suggestion
Durable Objects is designed to scale horizontally across many objects, but note that each individual object is inherently single-threaded.

An individual object has a soft limit of 1000 requests per second. A simple get on a small value that directly returns the response may realize a higher request throughput.

Correspondingly, an object that (for example) serializes and/or deserializes large JSON values or performs larger `list()` operations may be more limited in terms of request throughput. An object that receives too many requests will, after attempting to queue them, return an [overloaded](/durable-objects/platform/troubleshooting/#error-durable-object-is-overloaded) error to the caller.
```

Durable Objects have been built such that the number of Objects in the system do not need to be limited. You can create and run as many separate Durable Objects as you want. The main limit to your usage of Durable Objects is the total storage limit per account. If you need more storage, contact your account team.

> 

Reviewer: ```suggestion
* <sup>1</sup> Durable Objects both bills and measures storage based on a gigabyte (1 GB = 1,000,000,000 bytes) and not a gibibyte (GiB).
```

---
pcx_content_type: concept
title: Limits
weight: 9
---

# Limits

> Durable Objects are only available on the Workers Paid plan.

Reviewer 2: ```suggestion
Durable Objects are only available on the [Workers Paid plan](/workers/platform/pricing/#workers).
```

| Feature                                    | Limit                                             |
| ------------------------------------------ | --------------------------------------------------|

> |  Number of objects                         | unlimited (within an account or of a given class) |

Reviewer 2: ```suggestion
|  Number of objects                         | Unlimited (within an account or of a given class) |
```

|  Websocket message size                    | 1 MiB (only for received messages)                |
|  CPU per request                           | 30s (including websocket messages)                |

Durable Objects can scale horizontally across many Objects. Each individual Object is inherently single-threaded.

An individual object has a soft limit of 1000 requests per second. A simple get on a small value that directly returns the response may realize a higher request throughput.

Correspondingly, an object that (for example) serializes and/or deserializes large JSON values or performs larger `list()` operations may be more limited in terms of request throughput. An object that receives too many requests will, after attempting to queue them, return an [overloaded](/durable-objects/platform/troubleshooting/#error-durable-object-is-overloaded) error to the caller.

Durable Objects have been built such that the number of Objects in the system do not need to be limited. You can create and run as many separate Durable Objects as you want. The main limit to your usage of Durable Objects is the total storage limit per account. If you need more storage, contact your account team.

> <sup>1</sup> Durable Objects both bills and measures storage based on a gigabyte (1 GB = 1,000,000,000 bytes) and not a gibibyte (GiB).

Reviewer 2: I think this is missing the limit request form? Look at the aside in this section
/workers/platform/limits/#account-plan-limits

|  Storage per account                       | 50 GB (can be raised by contacting Cloudflare) <sup>1</sup>   |
|  Storage per class                         | unlimited                                         |
|  Storage per Object                        | unlimited                                         |
|  Key size                                  | 2 KiB(2048 bytes)                                 |
|  Value size                                | 128 KiB (131072 bytes)                            |
|  Websocket message size                    | 1 MiB (only for received messages)                |
|  CPU per request                           | 30s (including websocket messages)                |

{{<Aside type="note">}}

You can request adjustments to limits that conflict with your project goals by contacting Cloudflare. To request an increase to a limit, complete the [Limit Increase Request Form](https://forms.gle/ukpeZVLWLnKeixDu7) and we will contact you with next steps.

{{</Aside>}}

Durable Objects can scale horizontally across many Durable Objects. Each individual Object is inherently single-threaded.

> An individual Object has a soft limit of 1000 requests per second. A simple get on a small value that directly returns the response may realize a higher request throughput.

Reviewer 2: ```suggestion
An individual Object has a soft limit of 1,000 requests per second. A simple get on a small value that directly returns the response may realize a higher request throughput.
```
